agent/recognition/binarymatch.py

```
@@ -0,0 +1,165 @@
+@AgentServer.custom_recognition("HSVShapeMatching")
+class HSVShapeMatching(CustomRecognition):
+    def analyze(
+        self,
+        context: Context,
+        argv: CustomRecognition.AnalyzeArg,
+    ) -> Union[CustomRecognition.AnalyzeResult, Optional[RectType]]:
+        try:
+            # 1. 解析参数
+            params = json.loads(argv.custom_recognition_param)
+            
+            # 兼容性处理：优先读取 target_node，如果没有则尝试读取 recognition
```

Ruff (0.15.0)

[warning] Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)